lib/bootsnap/load_path_cache/loaded_features_index.rb

```diff
@@ -117,8 +118,30 @@ def register(short, long = nil)
       STRIP_EXTENSION = /\.[^.]*?$/
       private_constant(:STRIP_EXTENSION)
 
-      def strip_extension(f)
-        f.sub(STRIP_EXTENSION, '')
+      # Might Ruby automatically search for this extension if
+      # someone tries to 'require' the file without it? E.g. Ruby
+      # will implicitly try 'x.rb' if you ask for 'x'.
+      #
+      # This is complex and platform-dependent, and the Ruby docs are a little
+      # handwavy about what will be tried when and in what order.
+      # So optimistically pretend that all known elidable extensions
+      # will be tried on all platforms, and that people are unlikely
+      # to name files in a way that assumes otherwise.
+      # (E.g. It's unlikely that someone will know that their code
+      # will _never_ run on MacOS, and therefore think they can get away
+      # with callling a Ruby file 'x.dylib.rb' and then requiring it as 'x.dylib'.)
+      #
+      # See <https://ruby-doc.org/core-2.6.4/Kernel.html#method-i-require>.
+      def is_extension_elidable(f)
+        f.end_with?('.rb', '.so', '.o', '.dll', '.dylib')
```

Comment:
My hunch is that this would be faster as `File.extname(f) =~ /^\.(rb|so|o|dll|dylib)$/`

Reply:
My (extremely unscientific 😊) microbenchmark of doing 10 million comparisons against a known filename suggests that using `end_with?` is faster. (About 26% of how long the regex match takes.)

This matches my intuition: in the happy case (`*.rb`), Ruby should only need to compare three characters, starting from a known offset. That's pretty hard to beat.

Out of curiosity I tried switching the extensions around in the regex version, and it doesn't seem to make a difference. So I'd say not being able to hint that the vast majority of files will be `*.rb` is hurting pretty badly there.

I also tried using `File.extname` vs. `end_with?` on the full filename, and using `File.extname` is significantly slower. I assume what's happening here is that File.extname ends up allocating memory for the new string, whereas `end_with?` can just compare it to a constant in-place.